socorro/external/es/crashstorage.py

```diff
@@ -168,13 +184,39 @@ def reconstitute_datetimes(processed_crash):
                 # not there? we don't care
                 pass
 
+    @staticmethod
+    def remove_bad_keys(raw_crash):
+        """Removes keys from the top-level of the dict that are bad
+
+        Good keys satisfy the following properties:
+
+        * have one or more characters
+        * are composed of a-zA-Z0-9_-
+
+        Anything else is a bad key and needs to be removed.
+
+        This modifies the crash in-place.
+
+        :arg dict raw_crash: raw crash data
+
+        """
+        if not raw_crash:
+            return
+
+        for key in list(raw_crash.keys()):
```

Comment:
It's a good practice in any language to do something like this when you're iterating over a thing and removing things from it. In Python 3, `keys()` returns a *view* of the list of keys, so you want to make a copy of the things you're iterating over before iterating over them and removing things.

I'll add a comment because you asked.